lib_lua/src/Makefile

```diff
@@ -18,7 +18,7 @@ ifeq ($(PLAT), sam7s)
 	MCU   = arm7tdmi
 	THUMB = -mthumb -mthumb-interwork
 	CC = arm-elf-gcc
-	CFLAGS= -mcpu=$(MCU) $(THUMB) -ggdb -Os -Wall $(MYCFLAGS) -DLUA_OPTIMIZE_MEMORY=0 -DLUA_USE_MKSTEMP=1
+	CFLAGS= -mcpu=$(MCU) $(THUMB) -ggdb -Os -Wall $(MYCFLAGS) -DLUA_OPTIMIZE_MEMORY=2 -DLUA_USE_MKSTEMP=1
```

Comment:
2 means really agressive memory opts.  0 means none.  This was part of our late night "OMFG MK1 works with the opts while MK2 doesn't... ok then lets just make it work that way for now" convo.